src/System.Text.Json/tests/Serialization/Value.ReadTests.cs

```diff
@@ -34,6 +34,31 @@ public static void ReadPrimitives()
             Assert.Equal("Hello", s2);
         }
 
+        [Fact]
+        public static void ReadPrimitivesWithWhitespace()
+        {
+            int i = JsonSerializer.Parse<int>(Encoding.UTF8.GetBytes(@" 1 "));
+            Assert.Equal(1, i);
+
+            int i2 = JsonSerializer.Parse<int>("2\t");
+            Assert.Equal(2, i2);
+
+            int? i3 = JsonSerializer.Parse<int?>("\r\nnull");
+            Assert.Null(i3);
+
+            long l = JsonSerializer.Parse<long>(Encoding.UTF8.GetBytes("\t" + long.MaxValue.ToString()));
+            Assert.Equal(long.MaxValue, l);
+
+            long l2 = JsonSerializer.Parse<long>(long.MaxValue.ToString() + " \r\n");
+            Assert.Equal(long.MaxValue, l2);
+
+            string s = JsonSerializer.Parse<string>(Encoding.UTF8.GetBytes(@"""Hello"" "));
+            Assert.Equal("Hello", s);
+
+            string s2 = JsonSerializer.Parse<string>(@"  ""Hello"" ");
```

Comment:
Thinking a bit more about using `continue` within the main loop (rather than breaking and doing one final `Read` with validation), I wonder what would happen if we had 2 primitive tokens next to each other. This should continue to throw (presumably the reader will catch such cases).

```C#
string s3 = JsonSerializer.Parse<string>(@"""Hello"" ""Hello""");
string s4 = JsonSerializer.Parse<string>(@"""Hello"" 42);
```

Reply:
The serializer should definitely throw.

In Json.NET there is a setting on the JSON reader to allow multiple content. Additional content can optionally be delimited by a comma

`{'hi':true}{'bye':true}` and `{'hi':true},{'bye':true}` could be read by the reader when the flag was true.

It is a Useful Feature and people like it. But you don't need it for 3.0.

Reply:
I've added two extra cases to Value.ReadTests ReadPrimitiveExtraBytesFail to cover that scenari @ahsonkhan.